### PR TITLE
[US-624] Create Letter with Header From Template

### PR DESCRIPTION
**Pull request by kellemNegasi**
This PR provides an example on how to create company letters with preset headers and footers.

letter_to_JohnDoe1

### CODE REVIEW

File: document/template-with-header/main.go [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	return letters, nil
+}
+
+// generateDoc creates a docx file using the template provided by `templatePath` and writes it to `outputName`.
+func generateDoc(templatePath string, l letter, outputName string) error {
+	templateDoc, err := document.OpenTemplate(templatePath)
+	if err != nil {
+		log.Fatalf("error opening Windows Word 2016 document: %s", err)
```
Comment (sampila) [inline document/template-with-header/main.go]: Probably just write
```
log.Fatalf("error opening file: %s", err)
```

File: document/template-with-header/main.go [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+	}
+	defer templateDoc.Close()
+
+	// We can confirm the existence of the styles by printing them one by one
+	// for _, s := range templateDoc.Styles.Styles() {
+	// 	fmt.Println("style", s.Name(), "has ID of", s.StyleID(), "type is", s.Type())
+	// }
```
Comment (sampila) [inline document/template-with-header/main.go]: Probably here can take "Normal" style and put into variable, so later can use that to set the style.

**Review by sampila — CHANGES_REQUESTED** [review]
Overall looks good, added minor comments

File: document/template-with-header/main.go [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+// letter represents a given message with letter content and receiver name.
+type letter struct {
+	Receiver   string   `json:"receiver"`
+	Paragraphs []string `json:"paragraphs"`
+}
+
+func main() {
+	letters, err := loadLetters("./data/letters.json")
+	if err != nil {
+		panic(err)
```
Comment (anovik) [inline document/template-with-header/main.go]: Maybe instead of panic - log error and return?

File: document/template-with-header/main.go [hunk 4]
```diff
@@ -0,0 +1,127 @@
...
+	}
+
+	for _, letter := range letters {
+		generateDoc("letter_template.docx", letter, fmt.Sprintf("./output/letter_to_%s.docx", letter.Receiver))
```
Comment (anovik) [inline document/template-with-header/main.go]: should we proceed possible error from generateDoc?

File: document/template-with-header/main.go [hunk 5]
```diff
@@ -0,0 +1,127 @@
+// Copyright 2017 FoxyUtils ehf. All rights reserved.
```
Comment (anovik) [inline document/template-with-header/main.go]: probably update year in copyright?

**Review by anovik — APPROVED** [review]

File: document/template-with-header/main.go [hunk 6]
```diff
@@ -0,0 +1,127 @@
...
+
+// generateDoc creates a docx file using the template provided by `templatePath` and writes it to `outputName`.
+func generateDoc(templatePath string, l letter, outputName string) error {
+	templateDoc, err := document.OpenTemplate(templatePath)
+	if err != nil {
+		return fmt.Errorf("error opening template file : %s", err)
+	}
+	defer templateDoc.Close()
+
+	var Normal string // style for the body of the letter
```
Comment (sampila) [inline document/template-with-header/main.go]: probably should be
```
var styleNormal string
```

File: document/template-with-header/main.go [hunk 7]
```diff
@@ -0,0 +1,127 @@
...
+	for _, s := range templateDoc.Styles.Styles() {
+		// fmt.Println("style", s.Name(), "has ID of", s.StyleID(), "type is", s.Type())
```
Comment (sampila) [inline document/template-with-header/main.go]: better to remove this and changes to 
```
Look for style named "normal" from document styles.
for _, s := range templateDoc.Styles.Styles() {
 ...
}
```

And later can remove the comments of
```
// style name taken from the doc.Styles.Styles() list
```

**Review by sampila — CHANGES_REQUESTED** [review]
added comments

**Review by sampila — APPROVED** [review]
Looks good to me

**Pull request merged**